FluidNC/src/Configuration/Parser.cpp

```diff
@@ -38,10 +40,50 @@ namespace Configuration {
         return result;
     }
 
+    char const* Parser::match(const char* pattern) {
+        if (token_.state != TokenState::Matching || token_.keyStart_ == nullptr) {
+            return nullptr;
+        }
+
+        regex_t reg;
+        if (regcomp(&reg, pattern, REG_EXTENDED | REG_NOSUB) != 0) {
+            log_error("Can not compile regex: " << pattern);
+            return nullptr;
+        }
+
+        size_t length = 0;
+        for (const char* pos = token_.keyStart_; pos != token_.keyEnd_; pos++) {
+            length++;
+        }
```

Comment:
Why not "size_t length = token_.keyEnd_ - token_.keyStart;" Am I missing something subtle?

Reply:
I believe it was just me not knowing what I am doing ;-/